src/hotspot/cpu/riscv/vm_version_riscv.hpp

```diff
@@ -158,31 +152,26 @@ class VM_Version : public Abstract_VM_Version {
       return RVExtFeatures::current()->support_feature(_cpu_feature_index);
     }
     void enable_feature(int64_t value = 0) {
-      RVFeatureValue::enable_feature(value);
       RVExtFeatures::current()->set_feature(_cpu_feature_index);
     }
     void disable_feature() {
-      RVFeatureValue::disable_feature();
       RVExtFeatures::current()->clear_feature(_cpu_feature_index);
     }
+    int64_t value() { return enabled(); }
   };
 
   class RVNonExtFeatureValue : public RVFeatureValue {
-    bool _enabled;
+    int64_t _value;
+    static const int64_t DISABLED_VALUE = -1;
    public:
     RVNonExtFeatureValue(const char* pretty, int linux_bit_num, bool fstring) :
       RVFeatureValue(pretty, linux_bit_num, fstring),
-      _enabled(false) {
-    }
-    bool enabled()               { return _enabled; }
-    void enable_feature(int64_t value = 0) {
-      RVFeatureValue::enable_feature(value);
-      _enabled = true;
-    }
-    void disable_feature() {
-      RVFeatureValue::disable_feature();
-      _enabled = false;
+      _value(DISABLED_VALUE) {
     }
+    bool enabled() { return _value != DISABLED_VALUE; }
```

Comment:
Do we really need this `enabled()` method for non-extension features? I think this `enabled()` method should only apply to extension features.